ui/pages/confirmations/components/signature-request-original/signature-request-original.test.js

```diff
@@ -63,10 +27,21 @@ const props = {
   txData: {
     msgParams: {
       from: address,
-      data: MOCK_SIGN_DATA,
+      data: [
```

Comment:
Why the change in test data here? IIRC the data is stringified 🤔

Reply:
So I didn't change the way the data was formatted in the actual component for `eth_signTypedData` see [here](https://github.com/MetaMask/metamask-extension/pull/24756/files/2219e0ddbe92d95876d9bd3897462f6962389f5d#diff-0b1e3dca811d174401e32d4e02e4f70ebf269526c26c58f788cb7c3f0ee28ce0R132)... I agree this doesn't seem right. But as you can see [here](https://github.com/MetaMask/metamask-extension/blob/7a1191b33f1e5fac284fb7fb9ac1fa61102b0093/ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js#L224), this data is expected to be mappable. So what I've done here is preserve the existing behavior but make the test pass because of expectations embedded in the component that I didn't necessarily want to dig further into because of 👇 

A few things. This component is [technically deprecated](https://consensys.slack.com/archives/C03ETQA9EPK/p1721773925209699?thread_ts=1721768575.997869&cid=C03ETQA9EPK) and slated for removal shortly. I'm not clear on the original API method signature for `eth_signTypedData` (v1) since EIP-712 doesn't have a changelog...